### PR TITLE
Add the default constructor

### PR DESCRIPTION
**Pull request by neha-bhargava**

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs [hunk 1]
```diff
@@ -25,6 +25,11 @@ public abstract class AbstractAcquireTokenParameterBuilder<T>
     {
         internal IServiceBundle ServiceBundle { get; }
 
+        /// <summary>
+        /// Default constructor for AbstractAcquireTokenParameterBuilder
+        /// </summary>
```
Comment (trwalke) [inline src/client/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs:30]: should add [Obsolete("Please use the derived classes to construct this object.")]

Comment (trwalke) [reply src/client/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs:30]: @jmprieur thoughts?

Comment (jennyf19) [reply src/client/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs:30]: we don't need obsolete anymore.

---
In reply to: [480299373](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2022#discussion_r480299373) [](ancestors = 480299373)

Comment (neha-bhargava) [reply src/client/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs:30]: Ohh we dont?

Comment (jennyf19) [reply src/client/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs:30]: i don't think so. that causes the same issue. @jmprieur to make a final decision though.

Comment (jmprieur) [reply src/client/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs:30]: In this case, we don't need it as there are other constructors.
the reason why we read this constructor is to keep the binary compatibility.

**Review by jmprieur — CHANGES_REQUESTED** [review]
I propose to remove the obsolete attribute

**Comment by neha-bhargava** [comment]
> I propose to remove the obsolete attribute

Removed.

**Review by jmprieur — APPROVED** [review]

**Pull request merged**